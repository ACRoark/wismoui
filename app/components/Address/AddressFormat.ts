export const AddressFormats = ['block', 'street'] as const;

type AddressFormat = typeof AddressFormats[number];

export default AddressFormat;
