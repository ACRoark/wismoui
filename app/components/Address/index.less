@import (reference) '../../styles/themes/index.less';

@address-cls-prefix: ~'@{dte-prefix}-address';

.@{address-cls-prefix} {
  display: inline;
  font-style: normal;
  line-height: 1.375rem;
  white-space: pre-line;
}
