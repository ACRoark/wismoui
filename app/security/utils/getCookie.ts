const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
