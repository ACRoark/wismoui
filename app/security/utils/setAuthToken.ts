import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
