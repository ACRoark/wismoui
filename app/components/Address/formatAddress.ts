import { IAddress } from 'types';

import AddressFormat from './AddressFormat';

const formatAddress = (address: IAddress, format: AddressFormat): string => {
  const { city, line1, line2, state, zip } = address;

  const street = line1 + (line2 ? `, ${line2}` : '');

  if (format === 'street') {
    return street;
  }

  if (zip.indexOf('-') > -1) {
    const shortZip = zip.substring(0, zip.indexOf('-'));

    return `${street}\n${city}, ${state} ${shortZip}`;
  }

  return `${street}\n${city}, ${state} ${zip}`;
};

export default formatAddress;
