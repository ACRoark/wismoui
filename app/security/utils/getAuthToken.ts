import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
