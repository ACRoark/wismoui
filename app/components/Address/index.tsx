import React from 'react';

import AddressFormat from 'components/Address/AddressFormat';
import IAddress from 'types/IAddress';

import formatAddress from './formatAddress';

import './index.less';

interface IAddressProps {
  address: IAddress;
  format?: AddressFormat;
}

const Address: React.FC<IAddressProps> = (props: IAddressProps): React.ReactElement => {
  const { address, format = 'block' } = props;

  return <address className="dte-wismo-address">{formatAddress(address, format)}</address>;
};

export default Address;
