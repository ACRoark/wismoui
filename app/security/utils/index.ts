import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
