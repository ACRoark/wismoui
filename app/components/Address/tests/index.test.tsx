import { render } from 'enzyme';
import React from 'react';

import { createSnapshot } from 'testing/utils';
import IAddress from 'types/IAddress';

import Address from '..';
import AddressFormat, { AddressFormats } from '../AddressFormat';

const expectNoConsoleErrors = (format: AddressFormat, zipCode: string): void => {
  const spy = jest.spyOn(global.console, 'error');

  const address: IAddress = {
    city: 'Bloomfield',
    line1: '123 Main St',
    line2: 'Apt. A',
    state: 'MI',
    zip: zipCode,
  };

  render(<Address format={format} address={address} />);

  expect(spy).not.toHaveBeenCalled();
};

const expectAddressToMatchSnapshot = (format: AddressFormat, zipCode: string): void => {
  const address: IAddress = {
    city: 'Bloomfield',
    line1: '123 Main St',
    line2: 'Apt. A',
    state: 'MI',
    zip: zipCode,
  };

  const tree = createSnapshot(<Address format={format} address={address} />);

  expect(tree).toMatchSnapshot();
};

describe('Address Component', (): void => {
  const formats = AddressFormats;

  formats.map((format: AddressFormat): void => {
    it('should not log errors in console when rendered', (): void => {
      expectNoConsoleErrors(format, '48301');
    });

    it(`should render correctly when format is "${format}"`, (): void => {
      expectAddressToMatchSnapshot(format, '48301');
    });

    it(`should render correctly when format is "${format}", and zip has more than 5 digits`, (): void => {
      expectAddressToMatchSnapshot(format, '48301-1234');
    });
  });
});
