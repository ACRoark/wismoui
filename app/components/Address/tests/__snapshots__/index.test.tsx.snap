// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address Component should render correctly when format is "block" 1`] = `
<address
  className="dte-wismo-address"
>
  123 Main St, Apt. A
Bloomfield, MI 48301
</address>
`;

exports[`Address Component should render correctly when format is "block", and zip has more than 5 digits 1`] = `
<address
  className="dte-wismo-address"
>
  123 Main St, Apt. A
Bloomfield, MI 48301
</address>
`;

exports[`Address Component should render correctly when format is "street" 1`] = `
<address
  className="dte-wismo-address"
>
  123 Main St, Apt. A
</address>
`;

exports[`Address Component should render correctly when format is "street", and zip has more than 5 digits 1`] = `
<address
  className="dte-wismo-address"
>
  123 Main St, Apt. A
</address>
`;
